### ISSUE TITLE
Dictionary of metrics

### ISSUE THREAD

**Issue by bkj** [fchollet/keras#5660]
I'm trying to pass a dictionary of metrics in `Keras==1.2.2` -- it doesn't appear to be working, unless I'm making some mistake:

```
model.compile(loss='categorical_crossentropy',   metrics={"named_accuracy" : 'accuracy'})
model.fit(..., verbose=1)
```
yields
```
Epoch 1/153
2708/2708 [==============================] - 0s - loss: 1.9458
Epoch 2/153
2708/2708 [==============================] - 0s - loss: 1.9383
Epoch 3/153
2708/2708 [==============================] - 0s - loss: 1.9288
Epoch 4/153
2708/2708 [==============================] - 0s - loss: 1.9177
Epoch 5/153
2708/2708 [==============================] - 0s - loss: 1.9055
...
```
whereas 
```
model.compile(loss='categorical_crossentropy',   metrics=['accuracy'])
model.fit(..., verbose=1)
```
yields
```
Epoch 40/153
2708/2708 [==============================] - 0s - loss: 1.0846 - acc: 0.5355
```
#### EDIT
A workaround is to edit `model.metric_names` to reflect the desired names.

**Comment by lomizandtyd** [comment]
Could you please paste some model code here? 
I'm ok with dictionary of metrics.

The dictionary of metrics is used to specify the output layer of the metrics, rather than change the metric's name.
